lib/Mojo/WebSocketProxy/Backend/JobAsync.pm

```diff
@@ -0,0 +1,101 @@
+package Mojo::WebSocketProxy::Backend::JobAsync;
+
+use strict;
+use warnings;
+
+use parent qw(Mojo::WebSocketProxy::Backend);
+
+no indirect;
+
+use IO::Async::Loop::Mojo;
+use Job::Async;
+use Scope::Guard qw(guard);
+
+use Log::Any qw($log);
+
+## VERSION
+
+sub new {
+    my ($class, %args) = @_;
+    # Avoid holding these - we only want the Job::Async::Client instance, and everything else
+    # should be attached to the loop (which sticks around longer than we expect to).
+    my $loop = delete $args{loop};
+    my $jobman = delete $args{jobman};
+
+    my $self = bless \%args, $class;
+
+    # We'd like to provide some flexibility for people trying to integrate this into
+    # other systems, so any combination of Job::Async::Client, Job::Async and/or IO::Async::Loop
+    # instance can be provided here.
+    $self->{client} //= do {
+        $jobman //= do {
+            # We don't hold a ref to this, since that might introduce unfortunate cycles
+            $loop //= do {
+                require IO::Async::Loop::Mojo;
+                IO::Async::Loop::Mojo->new;
+            };
+            $loop->add(
+                my $jobman = Job::Async->new
```

Comment:
This doesn't want to be `my`

Reply:
I think it needs to be, since this is the inner `do { ... }` block that will eventually be assigned via `$jobman //=`.

Was a bit dubious about this construct: I often use this pattern in my own code, but have grave doubts about the value of inflicting it on others. Pretty sure it would not be that hard to restructure to something more sensible.

Reply:
Ah; in that case rather than `$jobman //= do { ...` maybe use

    unless($jobman) {
        ... code that side-effects assignment into $jobman
    }

Reply:
Done e4027a1